### PR TITLE
feat(spanner): add databoost property for batch transactions

### PR DESCRIPTION
**Pull request by harshachinta**
Adding support for databoost enabled in PartitionedRead and PartitionedQuery

### CODE REVIEW

**Review by gauravpurohit06 — APPROVED** [review]
LGTM, Can we just re-check why the integration tests are passing ?

File: spanner/batch.go [hunk 1]
```diff
@@ -150,13 +150,14 @@ func (t *BatchReadOnlyTransaction) PartitionReadUsingIndexWithOptions(ctx contex
...
-		KeySet:         kset,
-		RequestOptions: createRequestOptions(readOptions.Priority, readOptions.RequestTag, ""),
+		Session:          sid,
+		Transaction:      ts,
+		Table:            table,
+		Index:            index,
+		Columns:          columns,
+		KeySet:           kset,
+		RequestOptions:   createRequestOptions(readOptions.Priority, readOptions.RequestTag, ""),
+		DataBoostEnabled: readOptions.DataBoostEnabled,
```
Comment (gauravpurohit06) [inline spanner/batch.go:160]: Add a one line comment: 
`If this is for a partitioned read and this field is set to true, the request will be
executed via Spanner independent compute resources. `

File: spanner/batch.go [hunk 2]
```diff
@@ -212,13 +213,14 @@ func (t *BatchReadOnlyTransaction) partitionQuery(ctx context.Context, statement
...
-		QueryOptions:   qOpts.Options,
-		RequestOptions: createRequestOptions(qOpts.Priority, qOpts.RequestTag, ""),
+		Session:          sid,
+		Transaction:      ts,
+		Sql:              statement.SQL,
+		Params:           params,
+		ParamTypes:       paramTypes,
+		QueryOptions:     qOpts.Options,
+		RequestOptions:   createRequestOptions(qOpts.Priority, qOpts.RequestTag, ""),
+		DataBoostEnabled: qOpts.DataBoostEnabled,
```
Comment (gauravpurohit06) [inline spanner/batch.go:223]: Add a one line comment:
`If this is for a partitioned query and this field is set to true, the request will be executed via Spanner independent compute resources. `

**Review by gauravpurohit06 — CHANGES_REQUESTED** [review]

File: spanner/batch.go (hunk 1 above)
Comment (harshachinta) [reply spanner/batch.go:160]: I have added it under ReadOptions
https://github.com/googleapis/google-cloud-go/pull/8152/files#diff-98c680ad0e9a6842d77932c6b37ee80a322d5b5e462a26a4f0154844ea4e8513R169

File: spanner/batch.go (hunk 2 above)
Comment (harshachinta) [reply spanner/batch.go:223]: i have added it under QueryOptions
https://github.com/googleapis/google-cloud-go/pull/8152/files#diff-98c680ad0e9a6842d77932c6b37ee80a322d5b5e462a26a4f0154844ea4e8513R169

**Review by gauravpurohit06 — APPROVED** [review]

File: spanner/transaction.go [hunk 3]
```diff
@@ -165,16 +165,20 @@ type ReadOptions struct {
 
 	// The request tag to use for this request.
 	RequestTag string
+
+	// If this is for a partitioned read and DataBoostEnabled field is set to true, the request will be executed via Spanner independent compute resources.
```
Comment (olavloite) [inline spanner/transaction.go:169]: nit: split this comment over two lines

Also add the following to the end of the comment: `Setting this option for regular read operations has no effect.`

File: spanner/transaction.go [hunk 4]
```diff
@@ -188,6 +192,9 @@ func (ro ReadOptions) merge(opts ReadOptions) ReadOptions {
 	if opts.RequestTag != "" {
 		merged.RequestTag = opts.RequestTag
 	}
+	if opts.DataBoostEnabled {
```
Comment (olavloite) [inline spanner/transaction.go:195]: Hmmm.... This means that you cannot override a `false` value in `ro`. I guess that is OK, as we don't allow 'unsetting' any of the other values through this method either. (No action needed, just thinking out loud....)

File: spanner/transaction.go [hunk 5]
```diff
@@ -357,16 +369,20 @@ type QueryOptions struct {
 
 	// The request tag to use for this request.
 	RequestTag string
+
+	// If this is for a partitioned query and DataBoostEnabled field is set to true, the request will be executed via Spanner independent compute resources.
```
Comment (olavloite) [inline spanner/transaction.go:373]: Same as above: split over two lines and add an extra sentence that it has no effect on regular reads.

**Review by olavloite — APPROVED** [review]

File: spanner/transaction.go (hunk 3 above)
Comment (harshachinta) [reply spanner/transaction.go]: Added Knut.

File: spanner/transaction.go (hunk 5 above)
Comment (harshachinta) [reply spanner/transaction.go]: Added

**Pull request merged**